yt_dlp/extractor/abematv.py

```diff
@@ -138,15 +138,16 @@ def _get_device_token(self):
 
         username, _ = self._get_login_info()
         AbemaTVBaseIE._USERTOKEN = username and self.cache.load(self._NETRC_MACHINE, username)
+        AbemaTVBaseIE._DEVICE_ID = str(uuid.uuid4())
```

Comment:
Why not just move the opener also up here?

```suggestion
        AbemaTVBaseIE._DEVICE_ID = str(uuid.uuid4())
        add_opener(self._downloader, AbemaLicenseHandler(self))

        username, _ = self._get_login_info()
        AbemaTVBaseIE._USERTOKEN = username and self.cache.load(self._NETRC_MACHINE, username)
```

Reply:
Oops, I initially did that and forgot to switch it back after fiddling with the codebase.

Reply:
Since the suggestion did not remove the `add_opener` call at the end of the function, I've manually applied the changes.